boa3/internal/model/builtin/method/minbytestringmethod.py

```diff
@@ -14,98 +11,89 @@ def __init__(self, arg_value: Optional[IType] = None):
-        )
-
-        while_condition = [
-            (Opcode.OVER, b''),
-            (Opcode.OVER, b'')
-        ]
-        while_condition_bytes_size = get_bytes_count(while_condition)
-        while_condition.append(OpcodeHelper.get_jump_and_data(Opcode.JMPGT, -while_bytes_size - while_condition_bytes_size))
-
-        while_else_body = [
-            (Opcode.DROP, b''),
-            (Opcode.DROP, b''),
-            (Opcode.OVER, b''),  # if len(str1) > len(str2) jmp to end
-            (Opcode.SIZE, b''),
-            (Opcode.OVER, b''),
-            (Opcode.SIZE, b''),
-            (Opcode.GT, b''),
-        ]
-
-        everything_after_while_check = while_body_after_compare + while_condition + while_else_body
-        jmp_while_check = OpcodeHelper.get_jump_and_data(Opcode.JMP, get_bytes_count(everything_after_while_check), True)
-        while_body_compare[-1] = jmp_while_check
-
-        everything_after_equal_check = (get_limit_index +
-                                        while_body_before_compare +
-                                        while_body_compare +
-                                        everything_after_while_check)
-        jmp_test_equal = OpcodeHelper.get_jump_and_data(Opcode.JMP, get_bytes_count(everything_after_equal_check), True)
-        test_if_are_equal[-1] = jmp_test_equal
-
-        final_test = [
-            (Opcode.JMPIF, Integer(4).to_byte_array(signed=True)),      # if condition is not True
-            (Opcode.DROP, b''),                                            # remove str1 <=> return str2
-            (Opcode.JMP, Integer(3).to_byte_array(signed=True)),        # else
-            (Opcode.NIP, b'')                                              # remove str2 <=> return str1
-        ]
-
-        return (
-            test_if_are_equal +
-            everything_after_equal_check +
-            final_test
-        )
+    def generate_internal_opcodes(self, code_generator):
+        from boa3.internal.model.builtin.builtin import Builtin
+        from boa3.internal.model.operation.binaryop import BinaryOp
+        from boa3.internal.neo.vm.opcode.Opcode import Opcode
+
+        # if (str1 != str2):
+        code_generator.duplicate_stack_item(2)
+        code_generator.duplicate_stack_item(2)
+        if_init_are_not_equal = code_generator.convert_begin_if()
+        code_generator.change_jump(if_init_are_not_equal, Opcode.JMPEQ)
+
+        #   min_len_str = min((len(str1), len(str2))
+        code_generator.duplicate_stack_item(2)
+        code_generator.convert_builtin_method_call(Builtin.Len, is_internal=True)
+        code_generator.duplicate_stack_item(2)
+        code_generator.convert_builtin_method_call(Builtin.Len, is_internal=True)
+        code_generator.insert_opcode(Opcode.MIN)    # Pensar nisso aqui depois
+        #   index = 0
+        code_generator.convert_literal(0)
+        code_generator.convert_begin_if()
```

Comment:
need to get the result of this method to call `code_generator.convert_end_if` to properly handle this if statement